test/functional/rpc_signmessagewithprivkey.py

```diff
@@ -0,0 +1,46 @@
+#!/usr/bin/env python3
+# Copyright (c) 2016-2019 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+"""Test RPC commands for signing messages with private key."""
+
+from test_framework.test_framework import BitcoinTestFramework
+from test_framework.util import (
+    assert_equal,
+    assert_raises_rpc_error,
+)
+
+class SignMessagesWithPrivTest(BitcoinTestFramework):
+    def set_test_params(self):
+        self.setup_clean_chain = True
+        self.num_nodes = 1
+        self.extra_args = [["-addresstype=legacy"]]
+
+    def run_test(self):
+        message = 'This is just a test message'
+
+        self.log.info('test signing with priv_key')
+        priv_key = 'cUeKHd5orzT3mz8P9pxyREHfsWtVfgsfDjiZZBcjUBAaGk1BTj7N'
+        address = 'mpLQjfK79b7CCV4VMJWEWAj5Mpx8Up5zxB'
+        expected_signature = 'INbVnW4e6PeRmsv2Qgu8NuopvrVjkcxob+sX8OcZG0SALhWybUjzMLPdAsXI46YZGb0KQTRii+wWIQzRpG/U+S0='
+        signature = self.nodes[0].signmessagewithprivkey(priv_key, message)
+        assert_equal(expected_signature, signature)
+        assert self.nodes[0].verifymessage(address, signature, message)
+
+        self.log.info('test parameter validity and error codes')
+        # signmessage(withprivkey) have two required parameters
+        for num_params in [0, 1, 3, 4, 5]:
```

Comment:
Is there a reason for running the loop here on L32 and L36 from 0 to 5?

Reply:
It skips 2 (`signmessagewithprivkey` has two required parameters). I think the intention here is to ensure the RPC returns an error if called with the wrong number of parameters.

Reply:
Alright! Makes sense then